[pytorch] redirect `fbcode//caffe2/c10:c10` to the OSS/conda version

jtbraun opened the pull request:
Summary: Some build rules want to consume c10 from an external source. Convert it to an alias (like other pytorch entry points), and redirect conda builds to use the `fbcode//conda/buck_integration/toolchains/third-party:torch-cpp` target.

Test Plan: CI.

Reviewed By: athmasagar

Differential Revision: D81518888

malfet reviewed (approved):
LGTM if CI is green

malfet commented on c10/build.bzl:
```diff
@@ -1,6 +1,6 @@
 def define_targets(rules):
```
Does Skylark supports optional arguments? If that is the case, I would suggest to use `name="c10"` argument and redefine it for internal builds

jtbraun committed:
[pytorch] redirect `fbcode//caffe2/c10:c10` to the OSS/conda version (#169004)

Summary:

Some build rules want to consume c10 from an external source. Convert it to an alias (like other pytorch entry points), and redirect conda builds to use the `fbcode//conda/buck_integration/toolchains/third-party:torch-cpp` target.

Test Plan: CI.

Reviewed By: mortzur

Differential Revision: D81518888

izaitsevfb commented:
@pytorchbot merge

izaitsevfb commented:
@pytorchbot merge
